File: After_Refactor_3/QuixBugs/max_sublist_sum.py
def max_sublist_sum(arr):
    """
    Returns the maximum sum of a contiguous subarray within the given array.

    Parameters:
    ----------
    arr: list
        the input array to find the maximum sum of a contiguous subarray

    Returns:
    -------
    max_so_far: int
        the maximum sum of a contiguous subarray within the given array
    """
    max_ending_here = max_so_far = 0

    for num in arr:
        max_ending_here = max(num, max_ending_here + num)
        max_so_far = max(max_so_far, max_ending_here)

    return max_so_far 

File: After_Refactor_3/QuixBugs(Java)/src/main/java/MaxSublistSum.java
package java_programs;

public class MaxSublistSum {
    public static int getMaxSublistSum(int[] arr) {
        int maxEndingHere = 0;
        int maxSoFar = 0;

        for (int num : arr) {
            maxEndingHere = Math.max(0, maxEndingHere + num);
            maxSoFar = Math.max(maxSoFar, maxEndingHere);
        }

        return maxSoFar;
    }
}